lxd/project/permissions.go

```diff
@@ -1050,3 +1053,43 @@ var aggregateLimitConfigValuePrinters = map[string]func(int64) string{
 		return units.GetByteSizeString(limit, 1)
 	},
 }
+
+// FilterUsedBy filters a UsedBy list based on project access
+func FilterUsedBy(r *http.Request, entries []string) []string {
+	// Shortcut for admins and non-RBAC environments.
+	if rbac.UserIsAdmin(r) {
+		return entries
+	}
+
+	// Filter the entries.
+	usedBy := []string{}
+	for _, entry := range entries {
+		projectName := Default
+
+		// Try to parse the query part of the URL.
+		fields := strings.Split(entry, "?")
+		if len(fields) > 1 {
+			query := fields[len(fields)-1]
+
+			v, err := url.ParseQuery(query)
+			if err != nil {
+				// Skip URLs we can't parse.
+				continue
+			}
+
+			val, ok := v["project"]
+			if ok && len(val) > 0 {
```

Comment:
@stgraber unless you specifically want to use the last `project` value, then there is the `values.Get(key)` function that returns the first one which might be easier here. https://golang.org/pkg/net/url/#Values.Get

Reply:
So the reason for not using Get is that I wanted to be able to distinguish an empty project name from no project name.

Not sure it really matters though.

Reply:
@stgraber I dont think we do so anywhere else AFAIK, if the project name is empty is is considered default I think.